<a href="https://colab.research.google.com/github/vg137/ibd-causal-multiomics/blob/main/01_gwas_to_genes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
# For accessing the uncompressed dataset

from google.colab import drive
drive.mount('/content/drive')

import pandas as pd

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [65]:
data_dir = "/content/drive/MyDrive/ibd_data"
gwas_data_subdir = "GCST90473841_harmonised"
gwas_filename = "GCST90473841.h.tsv.gz"

### Quick checks

In [73]:
# First few lines of the dataset
!gzip -dc $data_dir/$gwas_data_subdir/$gwas_filename | head -n 10

chromosome	base_pair_location	effect_allele	other_allele	beta	standard_error	effect_allele_frequency	p_value	rsid	ID	INFO	n	TEST	CHISQ	EXTRA	hm_coordinate_conversion	hm_code	variant_id
1	10475	T	A	-0.668968	0.330278	0.000151983	0.04281835711891	rs1404283754	1:10475:A:T	1	457287	ADD	4.10254	NA	lo	5	1_10475_A_T
1	10477	T	C	-0.0589447	0.174931	0.000508366	0.7361460735624317	rs1263055247	1:10477:C:T	1	457348	ADD	0.113542	NA	lo	10	1_10477_C_T
1	10482	A	G	-0.513176	0.447671	7.65106e-05	0.25166104752630275	rs1307108837	1:10482:G:A	1	457453	ADD	1.31406	NA	lo	10	1_10482_G_A
1	10482	C	G	0.701418	0.386351	0.000104929	0.06944804106280247	rs1307108837	1:10482:G:C	1	457453	ADD	3.29601	NA	lo	5	1_10482_G_C
1	10489	T	C	-0.182319	0.332796	0.000146452	0.5838012223915866	rs1227863484	1:10489:C:T	1	457487	ADD	0.300129	NA	lo	10	1_10489_C_T
1	10490	A	G	-0.0251759	0.110105	0.00133008	0.8191378615185961	rs565971701	1:10490:G:A	1	457490	ADD	0.0522826	NA	lo	10	1_10490_G_A
1	10490	T	G	-0.133365	0.365697	0.0001235

In [76]:
# Number of total lines (including the line with column names)
# !pigz -dc $data_dir/$gwas_data_subdir/$gwas_filename | wc -l

84399075


It takes time to find the number of lines using the above command. But the result is 84399075 lines (including one extra line with column names). This means that 84,399,074 different SNPs were studied.

In [77]:
gwas_chunk_iterator = pd.read_csv(f"{data_dir}/{gwas_data_subdir}/{gwas_filename}",
                      sep="\t",
                      compression="gzip",
                      chunksize=1_000_000)

In [69]:
for chunk in gwas_chunk_iterator:


   chromosome  base_pair_location effect_allele other_allele      beta  \
0           1               10475             T            A -0.668968   
1           1               10477             T            C -0.058945   
2           1               10482             A            G -0.513176   
3           1               10482             C            G  0.701418   
4           1               10489             T            C -0.182319   

   standard_error  effect_allele_frequency   p_value          rsid  \
0        0.330278                 0.000152  0.042818  rs1404283754   
1        0.174931                 0.000508  0.736146  rs1263055247   
2        0.447671                 0.000077  0.251661  rs1307108837   
3        0.386351                 0.000105  0.069448  rs1307108837   
4        0.332796                 0.000146  0.583801  rs1227863484   

            ID  INFO       n TEST     CHISQ  EXTRA hm_coordinate_conversion  \
0  1:10475:A:T     1  457287  ADD  4.102540    NaN     